### PR TITLE
Add accessibility label to show password checkbox

### LINKED ISSUE #15965: [Accessibility] "Show password" checkbox doesn't provide label or accessible name
### Steps to reproduce
1. Open admin account creation form
2. With screen reader (NVDA  or Orca) press tab key until give focus to the checkbox

### Expected behaviour
Tell us what should happen
Screen reader should announce "Show password" unchecked chebox
### Actual behaviour
Tell us what happens instead
Screen reader only announces that there is a checkbox, but users don't know this is for showing the password.
### Server configuration

**Operating system**:
Ubuntu 19.04, Mozilla Firefox and Orca screen reader. 
**Web server:**
Apache 2
**Database:**
MariaDB
**PHP version:**
PHP 7.2
**Nextcloud version:** (see Nextcloud admin page)
Nextcloud Master
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh installation 
**Where did you install Nextcloud from:**
Git repository

### PR DESCRIPTION
**Pull request by nesb0t**
Fixes #15965.

Added aria-label to show password checkbox. Tested and verified with NVDA.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: core/templates/installation.php [hunk 1]
```diff
@@ -49,7 +49,8 @@
 				value="<?php p($_['adminpass']); ?>"
 				autocomplete="off" autocapitalize="none" autocorrect="off" required>
 			<label for="adminpass" class="infield"><?php p($l->t( 'Password' )); ?></label>
-			<input type="checkbox" id="show" class="hidden-visually" name="show">
+			<input type="checkbox" id="show" class="hidden-visually" name="show"
+				   aria-label="Show password">
```
Comment (skjnldsv) [inline core/templates/installation.php]: Please use a translation method :)
```suggestion
				   aria-label="<?php p($l->t( 'Show password' )); ?>">
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: core/templates/installation.php [hunk 2]
```diff
@@ -49,7 +49,8 @@
...
 			<label for="show"></label>
```
Comment (jancborchardt) [inline core/templates/installation.php]: @nesb0t @nextcloud/accessibility I would have assumed it’s better to use the label text here? Or is the `aria-label` recommended?

Comment (MarcoZehe) [reply core/templates/installation.php]: > 
> 
> @nesb0t @nextcloud/accessibility I would have assumed it’s better to use the label text here? Or is the `aria-label` recommended?

No, if a label text is already there, and only the association is missing, use that. It also needs to be localized, so if some information is already available, no need to use aria-label. Checkbox inputs can have the same label association like text inputs, e. g. the for attribute of the label points to the ID of the checkbox input, and the association is done. No need for aria-label.

**Comment by jancborchardt** [comment]
Thank you @nesb0t for the contribution! :)

Based on the comments from @skjnldsv and @MarcoZehe, could I ask you to change the following:
- Instead of `aria-label` on the checkbox, the text can be put into the `label for="show"` block directly after.
- The `label for="show"` then also needs `class="hidden-visually"` just like the checkbox so it’s only visible for screenreaders.
- It should be translatable in the way @skjnldsv suggested.

Let us know if you need any help. ♥

File: core/templates/installation.php (hunk 2 above)
Comment (jancborchardt) [reply core/templates/installation.php]: Thanks a lot for clarifying, @MarcoZehe!

**Comment by rullzer** [comment]
Master is no 18 development. If this needs to go into 17 please follow the normal backport procedures after merging.

**Review by skjnldsv — APPROVED** [review]

**Comment by kesselb** [comment]
> Thank you @nesb0t for the contribution! :)
> 
> Based on the comments from @skjnldsv and @MarcoZehe, could I ask you to change the following:
> 
>     * Instead of `aria-label` on the checkbox, the text can be put into the `label for="show"` block directly after.
> 
>     * The `label for="show"` then also needs `class="hidden-visually"` just like the checkbox so it’s only visible for screenreaders.
> 
>     * It should be translatable in the way @skjnldsv suggested.
> 
> 
> Let us know if you need any help. ♥

@skjnldsv is this done? ;)

**Comment by skjnldsv** [comment]
God dammit

**Comment by kesselb** [comment]
Sorry :see_no_evil:

**Comment by skjnldsv** [comment]
> Sorry

All good ;)
I missed the other comments! It's indeed the thing to do!

**Comment by kesselb** [comment]
`The label for="show" then also needs class="hidden-visually" just like the checkbox so it’s only visible for screenreaders.`

Don't forget ;)

**Comment by skjnldsv** [comment]
god dammit

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**